adage/adage/settings.py

```diff
@@ -0,0 +1,147 @@
+"""
+Django settings for adage project.
+
+Generated by 'django-admin startproject' using Django 2.2.5.
+
+For more information on this file, see
+https://docs.djangoproject.com/en/2.2/topics/settings/
+
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/2.2/ref/settings/
+"""
+
+import os
+import yaml
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+# Read secrets from YAML file
+path = os.path.join(BASE_DIR, 'adage', 'config.yml')
+with open(path) as read_file:
+    config = yaml.full_load(read_file)
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = config.get('secret_key', 'django_secret_key')
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = config.get('debug', True)
+
+ALLOWED_HOSTS = config.get('allowed_hosts', [])
+
+# Application definition
+
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'rest_framework',
+    'django_filters',
+    'corsheaders',
```

Comment:
I enabled it mainly because the frontend and backend development will be separate for this project. Having CORS enabled will make the frontend development much easier. The frontend developer only need to know the URLs of the API. 

We haven't really decided exactly how the production server should be deployed. If the frontend page is served from the same host as the backend, we can disable CORS completely.  Does this make sense?

Reply:
Yea - at some point the default initial app name also became set to the broader project name:
https://docs.djangoproject.com/en/2.2/intro/tutorial01/#creating-a-project

It used to be that you would have a broad name and then you'd also customize the internal name in their docs (so you could have the `adage` project but, by convention, you wouldn't have an `adage` app as well). Now it looks like it's considered more standard to have project-specific code live in a directory with the same name. This means we could use `adage` here, which is more specific than `analyze`.

Reply:
Ah, I missed this comment in the discussion at line 49 above. If there's a way to just make everything fit under `adage`, then I'd say that's a good way to proceed for now. We can refactor into two "apps" (as I suggested above) down the road if we decide that would be helpful.

Reply:
I think I'm making CORS into a bigger issue than it really is. Dealing with CORS is an added configuration headache that I think we would want to simplify to help people get up and running on their own quickly, but as @dongbohu says we need it if we separate the front end and the back end.

Mainly my note above was a question to make sure I understood our target deployment architecture. We still have time to figure that out, but I'd argue that we want to nail it down sooner rather than later.